spec/factories/courses.rb

```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :course do
+    name { Faker::Educator.course_name }
+    description { Faker::Lorem.paragraph(sentence_count: 5) }
+    number_of_semesters { Faker::Number.within(range: 1..4) }
+    allocation { Faker::Number.within(range: 1..100) }
```

Comment:
nice really like that using random infomation, maybe it's useful to also add nil for the ones that could be nil.
that way if a test depends on it being present, it needs to be added into the test and makes it clearer what the test depends on.

```suggestion
    description { [nil, "", Faker::Lorem.paragraph(sentence_count: 5)].sample }
    number_of_semesters { Faker::Number.within(range: 1..4) }
    allocation { [nil,Faker::Number.within(range: 1..100)].sample }
```